### PR TITLE
[DMU] Implement Thran Portal

### PR DESCRIPTION
**Pull request by Alex-Vasile**

### CODE REVIEW

**Comment by Alex-Vasile** [comment]
> does this work with [[Urborg, Tomb of Yawgmoth]] or [[Chromatic Lantern]]?

I can add tests, but I am not sure why it wouldn't. What is the interaction you're worried about?

**Comment by theelk801** [comment]
the extra abilities they add should cost life to activate

**Comment by theelk801** [comment]
also you shouldn’t add the cost to the abilities because they don’t actually have the cost

**Comment by Alex-Vasile** [comment]
> also you shouldn’t add the cost to the abilities because they don’t actually have the cost

I don't understand what you mean

**Comment by theelk801** [comment]
the abilities don’t actually have “pay 1 life” added to them, there’s just an effect that requires you to pay an additional life when you activate, so [[Manascape Refractor]] should copy them abilities without needing to pay life

**Comment by Alex-Vasile** [comment]
I get you. Okay, will change that as well and base it off [[Brutal Suppression]]

**Comment by Alex-Vasile** [comment]
> the abilities don’t actually have “pay 1 life” added to them, there’s just an effect that requires you to pay an additional life when you activate, so [[Manascape Refractor]] should copy them without needing to pay life

done

File: Mage.Sets/src/mage/cards/t/ThranPortal.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        put(SubType.PLAINS, new WhiteManaAbility());
+        put(SubType.ISLAND, new BlueManaAbility());
+        put(SubType.SWAMP, new BlackManaAbility());
+        put(SubType.MOUNTAIN, new RedManaAbility());
+        put(SubType.FOREST, new GreenManaAbility());
+    }};
+
+    public ThranPortalManaAbilityContinousEffect() {
+        super(Duration.WhileOnBattlefield, Layer.TypeChangingEffects_4, SubLayer.NA, Outcome.Neutral);
+        staticText = "mana abilities of Thran Portal cost an additional 1 life to activate";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/t/ThranPortal.java:81]: don't forget to use `{this}`

File: Mage.Sets/src/mage/cards/t/ThranPortal.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+        return true;
+    }
+}
+
+class ThranPortalAdditionalCostEffect extends CostModificationEffectImpl {
+
+    ThranPortalAdditionalCostEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Benefit, CostModificationType.INCREASE_COST);
+        this.staticText = "mana abilities of Thran Portal cost an additional 1 life to activate";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/t/ThranPortal.java:141]: also here

**Review by theelk801 — CHANGES_REQUESTED** [review]
looks good to me apart from the two small notes I added

**Review by theelk801 — APPROVED** [review]

**Pull request merged**